crates/ruff_db/src/diagnostic/mod.rs

```diff
@@ -845,3 +845,16 @@ pub fn create_unsupported_syntax_diagnostic(
     diag.annotate(Annotation::primary(span).message(err.to_string()));
     diag
 }
+
+/// Creates a `Diagnostic` from a semantic syntax error.
+///
+/// See [`create_parse_diagnostic`] for more details.
+pub fn create_semantic_syntax_diagnostic(
+    file: File,
+    err: &ruff_python_parser::semantic_errors::SemanticSyntaxError,
+) -> Diagnostic {
+    let mut diag = Diagnostic::new(DiagnosticId::InvalidSyntax, Severity::Error, "");
+    let span = Span::from(file).with_range(err.range);
+    diag.annotate(Annotation::primary(span).message(err.to_string()));
+    diag
```

Comment:
I don't think it's good practice to create diagnostics without a main message. I suggest passing `err.to_string()` to the `Diagnostic::new` (instead of "") and instead use no message in the annotation (the annotation's message should describe the annotated part). 

We should make the same change for unsupported syntax and parse diagnostics but we can do this as a seprate PR (unless @BurntSushi thinks what we have now is better)